Source/Engine/McBopomofoLM.cpp

```diff
@@ -159,3 +121,52 @@ bool McBopomofoLM::phraseReplacementEnabled()
     return m_phraseReplacementEnabled;
 }
 
+void McBopomofoLM::setExternalConverterEnabled(bool enabled)
+{
+    m_externalConverterEnabled = enabled;
+}
+
+bool McBopomofoLM::externalConverterEnabled()
+{
+    return m_externalConverterEnabled;
+}
+
+void McBopomofoLM::setExternalConvrter(std::function<string(string)> externalConverter)
+{
+    m_externalConverter = externalConverter;
+}
+
+const vector<Unigram> McBopomofoLM::filterAndTransformUnigrams(const vector<Unigram> unigrams, const unordered_set<string>& excludedValues, unordered_set<string>& insertedValues)
```

Comment:
Perhaps comment that `insertedValues` is an in/out parameter and this method has the side-effect of updating that unordered set.